packages/remix-react/transition.ts

```diff
@@ -120,7 +120,7 @@ export type FetcherStates<TData = any> = {
     formData: FormData;
     formEncType: string;
     submission: ActionSubmission;
-    data: undefined;
+    data: TData | undefined;
```

Comment:
I think this was just always a type bug?  We absolutely kept data around](https://github.com/remix-run/remix/blob/remix%401.9.0/packages/remix-react/transition.ts#L676) on fetcher action resubmissions - so I'm not even sure how the types didn't yell at us there...

Reply:
Huh - yeah I'm not sure why this wasn't yelling at us before?

https://www.typescriptlang.org/play?#code/C4TwDgpgBAYhwGMAWEBOBlYBDYEDOUAvFAN4BQUUAkgCYA2EAXKRZVHtrswEQCW9EbgG5WlGjizMArgDsaEAGa8ZEGiMoBfdVHRSARgFtewYMoDmAQQSmA9jObk27Tkyjc8+oyfPDRUcdjScorKqtparAAidq6ObBw4rnwCvk4BklBYMiDhIhFkEAAeYDaowFCgkLDwyGhE1YgoGC54ANoA1hAgNgoNtc2JeAC6ImQM5TQxcI1ozNP99XEJXFAA5PwMqwA0rOk8eDYG8Eg+ZPnjUCoA7vNNczVNmIOt3LqGxqYylta8dtxDi1YyySHne3i+3B2lAA9NCoAB1JAgKC8PAyVblYAnAhZKBoVClAD8lDeXk+31sMigqAgAEcpLwaQQ9rJ5EoVDRCbsJMxJipbmgAHTpKAAH1FUFZIQ5ZxEQA

Reply:
The issue is that `TData` is `any`, which TS treats as compatible with `undefined`. If we more accurately typed `TData` as `unknown` (or whatever it _actually_ is), then the error would show up in TS.